barman/retention_policies.py

```diff
@@ -185,6 +210,16 @@ def _backup_report(self, source):
                 i = i + 1
             else:
                 report[bid] = BackupInfo.NONE
+
+            if backups[bid].has_children:
+                status = report[bid]
+                if report[bid] in (BackupInfo.KEEP_STANDALONE, BackupInfo.KEEP_FULL):
```

Comment:
```suggestion
                if status in (BackupInfo.KEEP_STANDALONE, BackupInfo.KEEP_FULL):
```

We can skip the lookup in this line 😆